cmd/podman/system/service.go

```diff
@@ -125,6 +125,15 @@ func resolveAPIURI(uri []string) (string, error) {
 
 	switch {
 	case len(uri) > 0 && uri[0] != "":
+		// We want to check if the user is requesting a TCP address.
+		// If so, warn that this is insecure.
+		// Ignore errors here, the actual backend code will handle them
+		// better than we can here.
+		parsedURI, err := url.Parse(uri[0])
+		if err == nil && parsedURI.Scheme == "tcp" {
+			logrus.Warnf("Using the Podman API service with TCP sockets is not recommended, please see `podman system service` manpage for details")
+		}
+
```

Comment:
just move this down the stack where we already parse the uri and now it is tcp

Reply:
I was debating this - we usually don't like printing warnings out of library code. Happy to move if this is consensus.